dotCMS/src/main/java/com/dotcms/storage/DataBaseStoragePersistenceAPIImpl.java

```diff
@@ -0,0 +1,652 @@
+                    throw new DotRuntimeException(ex);
+                }
+            }
+
+            throw new DotRuntimeException(e);
+        } finally {
+
+            if (null != connection) {
+
+                try {
+                    connection.setAutoCommit(autocommit);
+                } catch (SQLException e) {
+
+                    Logger.error(this, e.getMessage(), e);
+                } finally {
+
+                    CloseUtils.closeQuietly(connection);
+                }
+            }
+        }
+
+        return result;
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @param groupName {@link String} group name
+     */
+    @Override
+    public boolean existsGroup(final String groupName) throws DotDataException {
+        final String groupNameLC = groupName.toLowerCase();
+        final MutableBoolean exists = new MutableBoolean(false);
+        try (Connection connection = getConnection()) {
+            final Number results = Try
+                    .of(() -> {
+                                final List<Map<String, Object>> result = new DotConnect()
+                                        .setSQL("SELECT count(*) as x FROM storage_group WHERE group_name = ?")
+                                        .addParam(groupNameLC)
+                                        .loadObjectResults(connection);
+                                return (Number) result.get(0).get("x");
+                            }
+                    ).getOrElse(0);
+            exists.setValue(results.intValue() > 0);
+        } catch (SQLException e) {
+            throw new DotDataException(e.getMessage(), e);
+        }
+        return exists.booleanValue();
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @param groupName {@link String}
+     * @param objectPath {@link String}
+     */
+    @Override
+    public boolean existsObject(final String groupName, final String objectPath)
+            throws DotDataException {
+        final String groupNameLC = groupName.toLowerCase();
+        final String objectPathLC = objectPath.toLowerCase();
+        final MutableBoolean exists = new MutableBoolean(false);
+
+        try (Connection connection = getConnection()) {
+            final Number results = Try
+                    .of(() -> {
+                                final List<Map<String, Object>> result = new DotConnect()
+                                        .setSQL("SELECT count(*) as x FROM storage WHERE group_name = ? AND path = ?")
+                                        .addParam(groupNameLC).addParam(objectPathLC)
+                                        .loadObjectResults(connection);
+                                return (Number) result.get(0).get("x");
+                            }
+                    ).getOrElse(0);
+            exists.setValue(results.intValue() > 0);
+        } catch (SQLException e) {
+            throw new DotDataException(e.getMessage(), e);
+        }
+        ;
```

Comment:
Codacy found an issue: [An empty statement (semicolon) not part of a loop](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6220272)